plugin-cgroupv1/src/cgroupv1.rs

```diff
@@ -37,47 +41,100 @@ impl Metrics {
-    cpu_usage_file.read_to_string(&mut buffer)?;
-    let total_cpu_time = buffer.trim().parse::<u64>()?;
-    buffer.clear();
-    let memory_usage_file = &mut job_source.oar2_metric_file.cgroup_memory_file;
-    memory_usage_file.rewind()?;
-    memory_usage_file.read_to_string(&mut buffer)?;
-    let memory_usage_u64 = buffer.trim().parse::<u64>()?;
+pub struct Cgroupv1Probe {
+    metrics: Metrics,
 
-    let mut measurement_point_vector: Vec<MeasurementPoint> = Vec::new();
-    measurement_point_vector.push(
-        MeasurementPoint::new(
-            timestamp,
-            job_source.memory_metric,
-            Resource::LocalMachine,
-            job_source.oar2_metric_file.memory_file_path.clone(),
-            memory_usage_u64,
-        )
-        .with_attr("job_id", job_source.oar2_metric_file.job_id.clone()),
-    );
+    cpu_time_delta_consumer: Option<ResourceConsumer>,
+    cpu_time_delta_file: Option<File>,
+    cpu_time_delta_counter_diff: Option<CounterDiff>,
 
-    let cpu_time_delta = match job_source.cpu_metric_counter_diff.update(total_cpu_time) {
-        CounterDiffUpdate::FirstTime => None,
-        CounterDiffUpdate::Difference(diff) => Some(diff),
-        CounterDiffUpdate::CorrectedDifference(diff) => Some(diff),
-    };
-    if let Some(cpu_time_delta_value) = cpu_time_delta {
-        measurement_point_vector.push(
-            MeasurementPoint::new(
+    memory_usage_consumer: Option<ResourceConsumer>,
+    memory_usage_file: Option<File>,
+}
+
+impl Cgroupv1Probe {
+    pub fn new(
+        metrics: Metrics,
+        cpuacct_usage_filepath: Option<String>,
+        memory_usage_in_bytes_filepath: Option<String>,
+    ) -> Result<Self, anyhow::Error> {
+        let mut probe = Self {
+            metrics: metrics,
+            cpu_time_delta_consumer: None,
+            cpu_time_delta_file: None,
+            cpu_time_delta_counter_diff: None,
+            memory_usage_consumer: None,
+            memory_usage_file: None,
+        };
+        if let Some(filepath) = cpuacct_usage_filepath {
+            probe.cpu_time_delta_consumer = Some(ResourceConsumer::ControlGroup {
+                path: filepath.clone().into(),
+            });
+            probe.cpu_time_delta_file = Some(File::open(filepath)?);
+            probe.cpu_time_delta_counter_diff = Some(CounterDiff::with_max_value(u64::MAX.into()));
+        }
+        if let Some(filepath) = memory_usage_in_bytes_filepath {
+            probe.memory_usage_consumer = Some(ResourceConsumer::ControlGroup {
+                path: filepath.clone().into(),
+            });
+            probe.memory_usage_file = Some(File::open(filepath)?);
+        }
+        Ok(probe)
+    }
+
+    pub fn collect_measurements(
+        &mut self,
+        timestamp: Timestamp,
+        additional_attrs: &HashMap<String, String>,
+    ) -> Result<Vec<MeasurementPoint>, PollError> {
+        let mut measurement_points: Vec<MeasurementPoint> = Vec::new();
```

Comment:
```suggestion
        let mut measurement_points = Vec::<MeasurementPoint>::new();
```
IDK if it is the best but I remember this kind of trick to not always display on the left side the type of variable

Reply:
Maybe don't change your code

Reply:
imo both are equivalent in term of readibility but I will change it as you said .